src/main/java/com/ldtteam/structures/client/BlueprintHandler.java

```diff
@@ -62,26 +52,49 @@ public static BlueprintHandler getInstance()
      * @param blueprint the wayPointBlueprint to draw.
      * @param pos       its position.
      */
-    public void draw(
-        final Blueprint blueprint,
-        final BlockPos pos,
-        final MatrixStack stack,
-        final float partialTicks)
+    public void draw(final Blueprint blueprint, final BlockPos pos, final MatrixStack stack, final float partialTicks)
     {
         if (blueprint == null)
         {
             Log.getLogger().warn("Trying to draw null blueprint!");
             return;
         }
+        Minecraft.getInstance().getProfiler().startSection("struct_render_cache");
 
-        try
+        final int blueprintHash = blueprint.hashCode();
+        final SoftReference<BlueprintRenderer> rendererRef = rendererCache.get(blueprintHash);
+        final BlueprintRenderer renderer;
+        final long now = System.nanoTime();
```

Comment:
use System.miliseconds unless you really need nanoTime precision, nanoTime is relatively slow